### PR TITLE
T1641 post filtering

### PR DESCRIPTION
**Pull request by rjmackay**
[Review on Reviewable](https://reviewable.io/reviews/ushahidi/platform/276)

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="http://app.review.ninja/ushahidi/platform/pull/276" target="_blank">ReviewNinja

File: application/classes/Ushahidi/Repository/Post.php [hunk 1]
```diff
@@ -297,6 +297,73 @@ protected function setSearchConditions(SearchData $search)
...
+      $stages = $search->current_stage;
+      if (!is_array($stages)) {
+        $stages = explode(',', $stages);
+      }
+
+      /**
+       * Here be dragons
+       * The purpose of this query is to return the set of posts which
+       * have current stage X. In this case, current stage X is actually the 
+       * the stage the post has NOT yet completed.
```
Comment (rjmackay) [inline application/classes/Ushahidi/Repository/Post.php]: Maybe mention its the **lowest priority** stage thats NOT completed

File: application/classes/Ushahidi/Repository/Post.php [hunk 2]
```diff
@@ -297,6 +297,73 @@ protected function setSearchConditions(SearchData $search)
...
+        ->where('form_stages.id', 'IN', $stages);
+
+      /**
+       * This is the master query, it collects all the posts
+       * missing stages that we are filtering by.
+       */
+      $sub = DB::select(array('posts.id','p_id'), array('form_stages.id', 'fs_id'), 'priority', 'forms.id')
+        ->from('posts')
+        ->join('forms')
+        //Here we join to the forms table based on the set of stage ids we are filtering by
```
Comment (rjmackay) [inline application/classes/Ushahidi/Repository/Post.php]: Minor: general comment style is to do ```// comment after a space```

**Comment by rjmackay** [comment]
I think this mostly looks good. However the Travis build failed.. can you fix that and update?

**Comment by rjmackay** [comment]
Looks good. Can you squash this down to a single commit with a description of what changed? Then we can merge

**Comment by billdoran** [comment]
Ok, what is the prefered way to squash?

**Comment by rjmackay** [comment]
1. Pretty sure you need to fix up indenting (use tabs not spaces)
2. If you run "git rebase -i master" then set all but the first commit to "squash" everything should get combined into a single commit
Edit the commit message to something sensible. Then "git push origin T1641-post-filtering -f"
Once you're sure this PR contains ONLY the required changes you can hit Merge

**Pull request closed without merging**